general/management/commands/report_count_statuses.py

```diff
@@ -74,153 +80,130 @@ def handle(self,  *args, **options):
             'User.num_posts': list(),
         }
 
+        # IMPLEMENTATION NOTE: on the code below we iterate multiple times on Sounds, Packs and Users tables.
+        # This is done in this way because due to our DB structure and the way that Django ORM works, if we annotate
+        # counts from different tables in a single queryset (annotations that require SQL JOINs with more than one
+        # table), the result we obtain is not the count that we want (see https://code.djangoproject.com/ticket/10060)
+        # The easier solution for us is to do the queries individually for each kind of "count" that we want to
+        # annotate.
+
         # Sounds
         total = Sound.objects.all().count()
-        print "Iterating over existing %i sounds..." % total,
-
-        annotations = {
-            'real_num_comments': Count('comments'),
-            'real_num_ratings': Count('ratings'),
-            'real_avg_rating': Avg('ratings__rating'),
-        }
-        if not options['skip-downloads']:
-            annotations['real_num_downloads'] = Count('downloads')
-
-        for count, sound in enumerate(Sound.objects.all().annotate(**annotations).iterator()):
-
-            needs_save = False
 
-            # Check num_comments
+        # Look at number of comments
+        for count, sound in enumerate(Sound.objects.all().annotate(real_num_comments=Count('comments')).iterator()):
             real_num_comments = sound.real_num_comments
```

Comment:
The only thing which annoys me thoroughly about this whole file is the usage of the "real_" prefix which isn't needed in any of the cases. The whole thing would be so much more readable without that prefix. If you find-replace "real_" with nothing I think this all still works just fine and is a lot more readable.

Reply:
The objects I'm updating already have a property with the same name without the prefix `real_`. The point here is that I'm fixing those which are wrong, assuming that there will be mismatches in the DB. Therefore to me it makes sense to add the `real_` prefix as these are the counts which have been computed "right now" and that have the correct value. Also to me it makes it readable because "if real number of apples is not equal to number of apples then X". Also I think that removing the `real_` prefix will break the code as then the annotated property will collide with the actual property. 

TL/DR: I won't change it :)